extension/app/ts/simulation/services/EthereumJSONRpcRequestHandler.ts

```diff
@@ -0,0 +1,27 @@
+import { JsonRpcResponse } from '../../utils/wire-types.js'
+
+export type IEthereumJSONRpcRequestHandler = Pick<EthereumJSONRpcRequestHandler, keyof EthereumJSONRpcRequestHandler>
+export class EthereumJSONRpcRequestHandler {
+	private nextRequestId: number = 1
+	private endpoint: string
+
+	constructor(endpoint: string) {
+		this.endpoint = endpoint
+    }
+
+	public readonly jsonRpcRequest = async (method: string, params: readonly unknown[]) => {
```

Comment:
Consider making these into a single object parameter and having the type be a union of all known Ethereum JSON-RPC types like:
```suggestion
	type EthereumJsonRpcRequest = EthereumGetBlockByNumberRequest | EthereumGetBlockByHashRequest | EthereumGetTransactionReceipt | ...
	...
	public readonly jsonRpcRequest = async (request: EthereumJsonRpcRequest) => {
```
This way, you can do serialization inside this function instead of having to do it prior to calling this function everywhere.

If you *really* want to go crazy you can do some complicated type stuff to make it so you can even have this function parse the result based on the expected result type by having a mapping of request to response types somewhere.  This requires quite a bit of effort to setup and at least one sacrifice to the TypeScript Gods, but it can really improve the ergonomics of the library going forward.

Reply:
yeah, I added input value checking as that was easy to do and has quite significant improvements. The return values is a good idea, but Its bit too much work for now, I created issue about it